server/services/interventionsService.test.ts

```diff
@@ -1290,6 +1290,233 @@ pactWith({ consumer: 'Interventions UI', provider: 'Interventions Service' }, pr
+          body: {
+            numberOfSessions: 4,
+          },
+        },
+        willRespondWith: {
+          status: 201,
+          body: Matchers.like({
+            id: draftActionPlanId,
+            referralId: '81d754aa-d868-4347-9c0f-50690773014e',
+            numberOfSessions: 4,
+            activities: [
+              {
+                id: '91e7ceab-74fd-45d8-97c8-ec58844618dd',
+                description: 'Attend training course',
+                desiredOutcomeId: '301ead30-30a4-4c7c-8296-2768abfb59b5',
+              },
+              {
+                id: 'e5755c27-2c85-448b-9f6d-e3959ec9c2d0',
+                description: 'Attend session',
+                desiredOutcomeId: '65924ac6-9724-455b-ad30-906936291421',
+              },
+            ],
+            submittedBy: null,
+            createdAt: '2020-12-07T20:45:21.986389Z',
+          }),
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        },
+      })
+
+      const draftActionPlan = await interventionsService.updateDraftActionPlan(token, draftActionPlanId, {
+        numberOfSessions: 4,
+      })
+      expect(draftActionPlan.id).toBe(draftActionPlanId)
+      expect(draftActionPlan.referralId).toBe('81d754aa-d868-4347-9c0f-50690773014e')
+      expect(draftActionPlan.numberOfSessions).toBe(4)
+    })
+  })
+
+  describe('submitDraftActionPlan', () => {
+    const submittedActionPlan = {
+      id: '486ba46a-0b57-46ab-82c0-d8c5c43710c6',
+      actionPlanFields: {
+        referralId: '81d754aa-d868-4347-9c0f-50690773014e',
+        numberOfSessions: 4,
+        activities: [
+          {
+            id: '91e7ceab-74fd-45d8-97c8-ec58844618dd',
+            description: 'Attend training course',
+            desiredOutcomeId: '301ead30-30a4-4c7c-8296-2768abfb59b5',
+            createdAt: '2020-12-07T20:45:21.986389Z',
+          },
+          {
+            id: 'e5755c27-2c85-448b-9f6d-e3959ec9c2d0',
+            description: 'Attend session',
+            desiredOutcomeId: '65924ac6-9724-455b-ad30-906936291421',
+            createdAt: '2020-12-07T20:47:21.986389Z',
+          },
+        ],
+      },
+      submittedBy: {
+        username: 'BERNARD.BEAKS',
+        userId: '555224b3-865c-4b56-97dd-c3e817592ba3',
+        authSource: 'delius',
+      },
+      submittedAt: '2020-12-08T20:47:21.986389Z',
+    }
+
+    beforeEach(async () => {
+      await provider.addInteraction({
+        state: 'a draft action plan with ID 6e8dfb5c-127f-46ea-9846-f82b5fd60d27 exists and is ready to be sent',
+        uponReceiving: 'a POST request to send the draft action plan with ID 6e8dfb5c-127f-46ea-9846-f82b5fd60d27',
+        withRequest: {
+          method: 'POST',
+          path: '/draft-action-plan/6e8dfb5c-127f-46ea-9846-f82b5fd60d27/submit',
+          headers: { Accept: 'application/json', Authorization: `Bearer ${token}` },
+        },
+        willRespondWith: {
+          status: 201,
```

Comment:
it responds with the full action plan with created at/by and submitted at/by details populated